diff --git a/tensorflow/BUILD b/tensorflow/BUILD
index 7c3b6e59faf..93132a50e57 100644
--- a/tensorflow/BUILD
+++ b/tensorflow/BUILD
@@ -89,6 +89,7 @@ PACKAGE_STATIC_DEPS = [
     "@local_execution_config_platform//:__subpackages__",
     "@mkl_dnn_acl_compatible//:__subpackages__",
     "@mkl_dnn_v1//:__subpackages__",
+    "@nccl_archive//:__subpackages__",
     "@org_sqlite//:__subpackages__",
     "@platforms//:__subpackages__",
     "@snappy//:__subpackages__",
