diff --git a/tensorflow/c/c_api.cc b/tensorflow/c/c_api.cc
index 1471348907d..0f7cf1b8e98 100644
--- a/tensorflow/c/c_api.cc
+++ b/tensorflow/c/c_api.cc
@@ -830,7 +830,7 @@ void TF_GraphGetTensorShape(TF_Graph* graph, TF_Output output, int64_t* dims,
 
 extern "C" {
 
-TF_OperationDescription* TF_NewOperationLocked(TF_Graph* graph,
+static TF_OperationDescription* TF_NewOperationLocked(TF_Graph* graph,
                                                const char* op_type,
                                                const char* oper_name)
     TF_EXCLUSIVE_LOCKS_REQUIRED(graph->mu) {
@@ -1080,7 +1080,7 @@ void TF_SetAttrValueProto(TF_OperationDescription* desc, const char* attr_name,
   status->status = Status::OK();
 }
 
-TF_Operation* TF_FinishOperationLocked(TF_OperationDescription* desc,
+static TF_Operation* TF_FinishOperationLocked(TF_OperationDescription* desc,
                                        TF_Status* status)
     TF_EXCLUSIVE_LOCKS_REQUIRED(desc->graph->mu) {
   Node* ret = nullptr;
diff --git a/tensorflow/c/c_api.h b/tensorflow/c/c_api.h
index 67207e84da0..b6f4c1792f2 100644
--- a/tensorflow/c/c_api.h
+++ b/tensorflow/c/c_api.h
@@ -259,14 +259,6 @@ TF_CAPI_EXPORT extern void TF_GraphGetTensorShape(TF_Graph* graph,
                                                   int64_t* dims, int num_dims,
                                                   TF_Status* status);
 
-// Creates a new operation - see `TF_NewOperation` for more details.
-//
-// The lock for `graph` must be held when calling this function.
-//
-// Unless implementing advanced behavior, like custom gradient functions, you
-// most likely need to call `TF_NewOperation` instead.
-TF_CAPI_EXPORT extern TF_OperationDescription* TF_NewOperationLocked(
-    TF_Graph* graph, const char* op_type, const char* oper_name);
 
 // Operation will only be added to *graph when TF_FinishOperation() is
 // called (assuming TF_FinishOperation() does not return an error).
@@ -419,15 +411,6 @@ TF_CAPI_EXPORT extern void TF_SetAttrValueProto(TF_OperationDescription* desc,
                                                 size_t proto_len,
                                                 TF_Status* status);
 
-// Adds this operation to the graph - see `TF_FinishOperation` for more details.
-//
-// The lock for `graph` must be held when calling this function.
-//
-// Unless implementing advanced behavior, like custom gradient functions, you
-// most likely need to call `TF_FinishOperation` instead.
-TF_CAPI_EXPORT extern TF_Operation* TF_FinishOperationLocked(
-    TF_OperationDescription* desc, TF_Status* status);
-
 // If this function succeeds:
 //   * *status is set to an OK value,
 //   * a TF_Operation is added to the graph,
