diff --git a/tensorflow/BUILD b/tensorflow/BUILD
index 19ee8000206..655c30af1e9 100644
--- a/tensorflow/BUILD
+++ b/tensorflow/BUILD
@@ -1162,6 +1162,7 @@ tf_cc_shared_library(
     ],
     dynamic_deps = select({
         "//tensorflow:macos": ["//tensorflow:libtensorflow_framework.%s.dylib" % VERSION],
+        "//tensorflow:windows": [],
         "//conditions:default": ["//tensorflow:libtensorflow_framework.so.%s" % VERSION],
     }),
     framework_so = [],
