diff -ruN tensorflow-2.6.0/tensorflow/BUILD tensorflow-2.6.0-visibility/tensorflow/BUILD
--- tensorflow-2.6.0/tensorflow/BUILD	2021-08-10 04:10:27.000000000 +0900
+++ tensorflow-2.6.0-visibility/tensorflow/BUILD	2021-08-30 11:18:31.089781754 +0900
@@ -37,7 +37,7 @@
 )
 load("@bazel_skylib//:bzl_library.bzl", "bzl_library")
 
-package(default_visibility = [":internal"])
+package(default_visibility = ["//visibility:public"])
 
 licenses(["notice"])
 
diff -ruN tensorflow-2.6.0/tensorflow/core/api_def/BUILD tensorflow-2.6.0-visibility/tensorflow/core/api_def/BUILD
--- tensorflow-2.6.0/tensorflow/core/api_def/BUILD	2021-08-10 04:10:27.000000000 +0900
+++ tensorflow-2.6.0-visibility/tensorflow/core/api_def/BUILD	2021-08-30 11:17:56.392705484 +0900
@@ -29,7 +29,7 @@
 alias(
     name = "base_api_def",
     actual = "//tensorflow/core/api_def/base_api:base_api_def",
-    visibility = ["//tensorflow:internal"],
+    visibility = ["//visibility:public"],
 )
 
 alias(
diff -ruN tensorflow-2.6.0/tensorflow/tools/api/lib/BUILD tensorflow-2.6.0-visibility/tensorflow/tools/api/lib/BUILD
--- tensorflow-2.6.0/tensorflow/tools/api/lib/BUILD	2021-08-10 04:10:27.000000000 +0900
+++ tensorflow-2.6.0-visibility/tensorflow/tools/api/lib/BUILD	2021-08-30 11:17:56.392705484 +0900
@@ -16,6 +16,7 @@
 tf_proto_library(
     name = "api_objects_proto",
     srcs = ["api_objects.proto"],
+    visibility = ["//visibility:public"],
 )
 
 py_library(
